lib/model/realm.dart

```diff
@@ -236,13 +272,61 @@ class RealmStoreImpl extends HasUserGroupStore with RealmStore {
     // That exists for deciding whether to offer the "Generate email address"
     // button, and if so then which users to offer in the dropdown;
     // it's predicting whether /api/get-stream-email-address would succeed.
-    if (_selfUserRole == UserRole.guest) {
-      final config = _groupSettingConfig(type, name);
-      if (!config.allowEveryoneGroup) return false;
+
+    final config = _groupSettingConfig(type, name);
+
+    if (_selfUserRole == UserRole.guest && !config.allowEveryoneGroup) {
+      return false;
     }
+
+    if (value == null) {
+      // The server doesn't know about the permission. *We* know about it
+      // (or presumably we wouldn't have called this method),
+      // and we know a reasonable default; use that.
+      return _hasPermissionByDefault(config, atDate: atDate);
+    }
+
     return selfInGroupSetting(value);
   }
 
+  bool _hasPermissionByDefault(
+    PermissionSettingsItem config, {
+    required DateTime atDate,
+  }) {
+    switch (config.defaultGroupName) {
+      case DefaultGroupNameUnknown():
+        // When we know about a permission, we should also know about the group
+        // we've said is the default value for it.
+        assert(false);
+        return true;
+      case PseudoSystemGroupName.streamCreatorOrNobody:
+        // TODO(#1102) implement
+        assert(() {
+          throw UnimplementedError();
+        }());
+        return true;
+      case SystemGroupName.everyoneOnInternet:
+      case SystemGroupName.everyone:
+        return true;
+      case SystemGroupName.members:
+        return _selfUserRole.isAtLeast(UserRole.member);
+      case SystemGroupName.fullMembers:
+        if (!_selfUserRole.isAtLeast(UserRole.member)) return false;
+        if (_selfUserRole == UserRole.member) {
+          return selfHasPassedWaitingPeriod(byDate: atDate);
+        }
+        return true;
```

Comment:
How about we instead just make this unimplemented, like `.streamCreatorOrNobody` above?

The difference being that in this case, we don't expect to ever add code that triggers this case. (I think we may have never had a permission that defaulted to "full members". If we ever add one in the future, we'd probably make the permission configurable and group-based from the beginning, so there would still never be a fallback to such a default for it.)